internal/graph/check.go

```diff
@@ -1411,10 +1411,18 @@ func (c *LocalChecker) checkTTU(parentctx context.Context, req *ResolveCheckRequ
 		if typesys.RecursiveTTUCanFastPath(objectTypeRelation, userType) {
 			resolver = c.recursiveTTUFastPath
 			span.SetAttributes(attribute.String("resolver", "recursivefastpathv1"))
-		} else if c.optimizationsEnabled && typesys.TTUCanFastPathWeight2(objectType, relation, userType, rewrite.GetTupleToUserset()) {
-			resolver = c.checkTTUFastPathV2
-			span.SetAttributes(attribute.String("resolver", "fastpathv2"))
+		} else if c.optimizationsEnabled {
```

Comment:
This seems off. In a case where `RecursiveTTUCanFastPath` is true, you wouldn't go into the new implementation. And even if you were, you are paying the cost twice. 

So if anything it should become:

if optimizationsEnabled {
  check fast paths v2
} else {
  the original ones
}

Reply:
And within the order of the optimizations to check, its more common to hit non recursive implementations, so that should come first.

Reply:
updated